### PR TITLE
pread error definition and anchor links for errors

### PR DESCRIPTION
**Pull request by sshanks-kx**
Errors now have anchor links e.g. 'domain' can now be linked directly using #domain.
Errors with a space, have the space changed to a dash e.g. "Could not initialize ssl" would be #Could-not-initialize-ssl

### CODE REVIEW

File: docs/basics/errors.md [hunk 1]
```diff
@@ -55,150 +62,180 @@ bad meta data in file
...
 badtail
 
 :   Incomplete transaction at end of file, get good (count;length) with ``-11!(-2;`:file)``
 
+[](){#binary-mismatch}
 binary mismatch
 
 :   Wrong process for [code profiler](../kb/profiler.md)
 
+[](){#can't}
```
Comment (JosieMD) [inline docs/basics/errors.md:75]: Is an apostrophe OK in an anchor link?

Comment (sshanks-kx) [reply docs/basics/errors.md:75]: had tested & link worked

File: docs/basics/errors.md [hunk 2]
```diff
@@ -55,150 +62,180 @@ bad meta data in file
...
 cond
 
 :   Even number of arguments to `$` (until V3.6 2018.12.06)
 
+[](){#conn}
 conn
 
 :   Too many connections. Max connections was 1022 prior to 4.1t 2023.09.15, otherwise the limit imposed by the operating system (operating system configurable for system/protocol).
 
+[](){#Could-not-initialize-ssl}
```
Comment (JosieMD) [inline docs/basics/errors.md]: Is this one really a capital letter when the rest are not?

Comment (sshanks-kx) [reply docs/basics/errors.md]: changed now to lowercase

**Pull request merged**